cpmpy/expressions/variables.py

```diff
@@ -334,6 +345,24 @@ def __ne__(self, other):
     def __abs__(self):
         return self
 
+    def __and__(self, other):
```

Comment:
Why not add this logic to the [Expression class in core.py](https://github.com/CPMpy/cpmpy/blob/9fba76cc89845deb99f01a0b396021cc7f708924/cpmpy/expressions/core.py#L206)?

Reply:
basically to avoid multiple checking of both sides and have a more, but can be moved to core.py if we also add a "self.is_bool()" check and throw the exception there (for cases of not boolean expressions, IntVars etc.).

Reply:
moving it there now